apps/api/src/app/messages/usecases/remove-messages-by-transactionId/remove-messages-by-transactionId.usecase.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { buildFeedKey, buildMessageCountKey, InvalidateCacheService } from '@novu/application-generic';
+import { MessageEntity, MessageRepository } from '@novu/dal';
+import { RemoveMessagesByTransactionIdCommand } from './remove-messages-by-transactionId.command';
+
+@Injectable()
+export class RemoveMessagesByTransactionId {
+  constructor(private messageRepository: MessageRepository, private invalidateCache: InvalidateCacheService) {}
+
+  async execute(command: RemoveMessagesByTransactionIdCommand) {
+    const subscriberIds = await this.messageRepository.find(
+      {
+        transactionId: command.transactionId,
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        deleted: false,
+        ...(command.channel && { channel: command.channel }),
+      },
+      '_subscriberId'
+    );
+
+    if (subscriberIds.length === 0) {
+      throw new NotFoundException('Invalid transactionId or channel');
+    }
+
+    for (const subscriberId of subscriberIds) {
```

Comment:
Maybe it will be worth checking creating a pipe to redis in order to reduce the number of calls to redis here, in case the subscriber count is big.

Reply:
Created a separate [ticket](https://linear.app/novu/issue/NV-2879/add-pipeline-support-to-handle-large-numbers-of-cache-invalidation) to handle large numbers of cache invalidation.